### PR TITLE
KIN-129 Category table view

### PR DESCRIPTION
**Pull request by FernandoJoachin**
[KIN-129](https://loopcrack.atlassian.net/browse/KIN-129) Medium [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

The category table was created along with its pagination

### Problem

The project requires a way to display existing categories on the categories page of the admin panel to allow users to view and interact with them.

### Solution

Create a view with a table to show the categories, as well as having buttons to create, edit or delete them.

### Requirements checklist

Include a list of requirements necessary to consider this pull request complete

- [x] A table with the existing categories must be displayed.
- [x] It must have a button for creating a new category.
- [x] Each register in the table must have the following: Name of the category, Categor, icon, Button to edit and Button to delete.
- [x] It must implement the private view pagination.

### Testing

- [x] Add more static categories in views/admin/categories/categories.php file to test performance.

## Demo

[](https://postimg.cc/v1Fc8bvV)
[](https://postimg.cc/VJMkV2cL)

[KIN-129]: https://loopcrack.atlassian.net/browse/KIN-129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by CarlosJCD — COMMENTED** [review]
Are we going to use the same parameters for responsive web design on the private view?

I am asking this because it seems that for devices that have 475px or less the layout starts to break, but I don't know how important is for the private view to have a responsive design.


Update 12/09/2023: The design its now responsive, excellent work!

File: app/Views/admin/categories/Categories.php [hunk 1]
```diff
@@ -1 +1,194 @@
...
+                                                </td>
+                                                <td>Icono.jpeg</td>
+                                                <td>Imagen.jpeg</td>
+                                                <td>
+                                                    <button type="button" class="btn btn-primary btn-icon waves-effect waves-light"><i class="ri-edit-2-fill"></i></button>
+                                                    <button type="button" class="btn btn-danger btn-icon waves-effect waves-light"><i class="ri-delete-bin-5-line"></i></button>
+                                                </td>
+                                            </tr>
+                                        </tbody>
+                                    </table>
```
Comment (LUGAMAFE) [inline app/Views/admin/categories/Categories.php:151]: Puedes añadir un tfoot porfavor @FernandoJoachin 

```suggestion
                                            </tr>
                                        </tbody>
                                        <tfoot>
                                            <tr>
                                               <th>ID</th>
                                               <th>Nombre</th>
                                               <th>Icono</th>
                                               <th>Imagen</th>
                                               <th>Acciones</th>
                                              </tr>
                                          </tfoot>
                                    </table>
```

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]
@FernandoJoachin te deje un monton de comentarios, pero no por que tu trabajo este mal, al contrario esta muy bien, pero creo que hay muchas cosas que podemos mejorar. 

Creo que me gustaria que platicaraas todos estos cambioes que te pedi con @Cristianpan y @Mauriciocr207

File: app/Views/admin/categories/Categories.php [hunk 2]
```diff
@@ -1 +1,265 @@
...
+                    <div class="card-header d-flex justify-content-between align-items-center flex-wrap">
+                        <h5 class="card-title mb-2 mb-md-0">Tabla de Categorías</h5>
+                        <div class="mt-2 mt-md-0 float-md-right">
+                            <button type="button" class="btn btn-success btn-label waves-effect waves-light rounded-pill">
+                                <i class="ri-add-circle-fill label-icon align-middle rounded-pill fs-16 me-2"></i>Agregar Categoría
+                            </button>
+                        </div>
+                    </div>
+                    <div class="card-body">
+                        <table id="category" class="table nowrap dt-responsive align-middle table-hover table-bordered" style="width:100%">
```
Comment (LUGAMAFE) [inline app/Views/admin/categories/Categories.php:31]:
```suggestion
                        <table id="categories-table" class="table nowrap dt-responsive align-middle table-hover table-bordered" style="width:100%">
```
change the name of the id

File: app/Views/templates/admin/dashboardTable.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+<head>
+
+    <?= $this->renderSection('title-meta') ?>
+
+    <?= $this->renderSection('css') ?>
+
+    <?= $this->include('partials/head-css') ?>
+
+    <style>
```
Comment (LUGAMAFE) [inline app/Views/templates/admin/dashboardTable.php:11]: Esta etiqueta style para que es?

File: public/assets/js/pages/category.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+document.addEventListener('DOMContentLoaded', function () {
+    $('#category').DataTable({
+        "language": {
+            "info": "Mostrando _START_ a _END_ de _TOTAL_ categorías",
+            "infoEmpty": "Mostrando 0 a 0 de 0 categorías",
+            "infoFiltered": "(filtrado de _MAX_ categorías totales)",
+            "lengthMenu": "Mostrar _MENU_ categorías",
+        }
+    });
```
Comment (LUGAMAFE) [inline public/assets/js/pages/category.js:9]:
```suggestion
    $('#category').DataTable({
        "language": {
            "info": "Mostrando _START_ a _END_ de _TOTAL_ categorías",
            "infoEmpty": "Mostrando 0 a 0 de 0 categorías",
            "infoFiltered": "(filtrado de _MAX_ categorías totales)",
            "lengthMenu": "Mostrar _MENU_ categorías",
        },
         columnDefs: [
            { orderable: false, responsivePriority: 1, targets: -1}
         ]   
    });
```
Añade esto porfavor @FernandoJoachin para mejorar la tabla

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]

File: app/Views/templates/admin/dashboardTable.php (hunk 3 above)
Comment (FernandoJoachin) [reply app/Views/templates/admin/dashboardTable.php:11]: No me di cuenta que deje eso

**Review by LUGAMAFE — APPROVED** [review]
muchas gracias @FernandoJoachin

**Review by CarlosJCD — APPROVED** [review]

**Review by CheloChidoone — APPROVED** [review]

**Pull request merged**